executor/split_test.go

```diff
@@ -366,6 +366,36 @@ func (s *testSplitIndex) TestSplitTable(c *C) {
 	}
 }
 
+func (s *testSplitIndex) TestCalculateIntBoundValue(c *C) {
+	tbInfo := &model.TableInfo{
+		Name: model.NewCIStr("t1"),
+		ID:   rand.Int63(),
+		Columns: []*model.ColumnInfo{
+			{
+				Name:         model.NewCIStr("c0"),
+				ID:           1,
+				Offset:       1,
+				DefaultValue: 0,
+				State:        model.StatePublic,
+				FieldType:    *types.NewFieldType(mysql.TypeLong),
+			},
+		},
+	}
+	ctx := mock.NewContext()
+	e := &SplitTableRegionExec{
+		baseExecutor: newBaseExecutor(ctx, nil, 0),
+		tableInfo:    tbInfo,
+		handleCols:   core.NewIntHandleCols(&expression.Column{RetType: types.NewFieldType(mysql.TypeLonglong)}),
+		lower:        []types.Datum{types.NewDatum(-9223372036854775808)},
+		upper:        []types.Datum{types.NewDatum(9223372036854775807)},
+		num:          4,
```

Comment:
What happens when `REGIONS 1` is given?

Reply:
I add some edge test. And in the Integration Testing, the mysql client get the result:
```
mysql> create table if not exists `t1` (`id` bigint);
Query OK, 0 rows affected (0.00 sec)

mysql> split table `t1` between (-9223372036854775808) and (9223372036854775807) regions 1;
ERROR 1105 (HY000): Split table `t1` region step value should more than 1000, step -1 is invalid
```